syntax = "proto3";

service TodoService{
  rpc GetTodo(GetTodoRequest) returns (GetTodoResponse); 
  rpc AddTodo(AddTodoRequest) returns (AddTodoResponse);
  rpc UpdateTodo(UpdateTodoRequest) returns (UpdateTodoResponse);
  rpc DeleteTodo(DeleteTodoRequest) returns (DeleteTodoResponse);
}

message AddTodoRequest {
  string task = 1;
  string status = 2;
}

message AddTodoResponse {
  string task = 1;
  string status = 2;
}

message GetTodoRequest {
  int32 id = 1;
}

message GetTodoResponse {
  int32 id = 1;
  string task = 2;
  string status = 3;
}

message UpdateTodoRequest {
  int32 id = 1;
  string task = 2;
  string status = 3;
}

message UpdateTodoResponse  {
  int32 id = 1;
  string task = 2;
  string status = 3;
}

message DeleteTodoRequest {
  int32 id = 1;
}

message DeleteTodoResponse {
  int32 id = 1;
}
